_15thnight/__init__.py

```diff
@@ -147,32 +145,33 @@ def dashboard():
         return render_template(
             'dashboard/admin.html',
             form=form,
+            cat_form=AddCategoryForm(),
             form_error=form_error,
             users=User.get_users(),
             alerts=Alert.get_alerts(),
+            categories=Category.all(),
             delete_user_form=DeleteUserForm(),
-            deleted_user=deleted_user
+            deleted_user=deleted_user,
+            deleted_category=deleted_category
         )
     elif current_user.role == 'advocate':
         # Advocate user, show alert form
         form = AlertForm()
         if request.method == 'POST' and form.validate_on_submit():
             alert = Alert(
                 description=form.description.data,
-                other=form.other.data,
-                shelter=form.shelter.data,
-                food=form.food.data,
-                clothes=form.clothes.data,
                 gender=form.gender.data,
                 age=form.age.data,
-                user=current_user
+                user=current_user,
+                categories=Category.get_by_ids(form.categories.data).all(),
             )
             alert.save()
-            users_to_notify = User.get_provider(
-                alert.food, alert.clothes, alert.shelter, alert.other
-            )
+
+            users_to_notify = User.query.join(User.categories) \
+                .filter(Category.id.in_(form.categories.data)) \
+                .distinct()
```

Comment:
This should be a single call that looks like:

`users_to_notify = User.get_providers(form.categories.data)`